node/src/arrow.ts

```diff
@@ -136,214 +196,304 @@ export function makeArrowTable (
+    const firstTable = new ArrowTable(columns)
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const batchesFixed = firstTable.batches.map(batch => new RecordBatch(opt.schema!, batch.data))
+    return new ArrowTable(opt.schema, batchesFixed)
+  } else {
+    return new ArrowTable(columns)
+  }
 }
 
-// Converts an Array of records into an Arrow Table, optionally applying an embeddings function to it.
-export async function convertToTable<T> (
-  data: Array<Record<string, unknown>>,
-  embeddings?: EmbeddingFunction<T>
-): Promise<ArrowTable> {
-  if (data.length === 0) {
-    throw new Error('At least one record needs to be provided')
+/**
+ * Create an empty Arrow table with the provided schema
+ */
+export function makeEmptyTable (schema: Schema): ArrowTable {
+  return makeArrowTable([], { schema })
+}
+
+// Helper function to convert Array<Array<any>> to a variable sized list array
+function makeListVector (lists: any[][]): Vector<any> {
+  if (lists.length === 0 || lists[0].length === 0) {
+    throw TypeError('Cannot infer list vector from empty array or empty list')
+  }
+  const sampleList = lists[0]
+  let inferredType
+  try {
+    const sampleVector = makeVector(sampleList)
+    inferredType = sampleVector.type
+  } catch (error: unknown) {
+    // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+    throw TypeError(`Cannot infer list vector.  Cannot infer inner type: ${error}`)
   }
 
-  const columns = Object.keys(data[0])
-  const records: Record<string, Vector> = {}
+  const listBuilder = makeBuilder({
+    type: new List(new Field('item', inferredType, true))
+  })
+  for (const list of lists) {
+    listBuilder.append(list)
+  }
+  return listBuilder.finish().toVector()
+}
 
-  for (const columnsKey of columns) {
-    if (columnsKey === 'vector') {
-      const vectorSize = (data[0].vector as any[]).length
-      const listBuilder = newVectorBuilder(vectorSize)
-      for (const datum of data) {
-        if ((datum[columnsKey] as any[]).length !== vectorSize) {
-          throw new Error(`Invalid vector size, expected ${vectorSize}`)
-        }
+// Helper function to convert an Array of JS values to an Arrow Vector
+function makeVector (values: any[], type?: DataType, stringAsDictionary?: boolean): Vector<any> {
+  if (type !== undefined) {
+    // No need for inference, let Arrow create it
+    return vectorFromArray(values, type)
+  }
+  if (values.length === 0) {
+    throw TypeError('makeVector requires at least one value or the type must be specfied')
+  }
+  const sampleValue = values.find(val => val !== null && val !== undefined)
+  if (sampleValue === undefined) {
+    throw TypeError('makeVector cannot infer the type if all values are null or undefined')
+  }
+  if (Array.isArray(sampleValue)) {
+    // Default Arrow inference doesn't handle list types
+    return makeListVector(values)
```

Comment:
Where do we handle list vs fixed size list?

Reply:
If the user wants fixed size list then either they supply a schema or they use `vectorColumns`.  Otherwise we will always interpret an `Array<Array<...>>`as a variable sized list.

Reply:
in our user guide, the default behavior is

```
data = [
  "vec": [[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
]
```
will it be interpreted as List<f32> instead of `FixedSizeList<f32>`?

Reply:
The examples I see in the user guide are all:

```
data = [
  { "vector": [1.0, 2.0, 3.0] },
  { "vector": [4.0, 5.0, 6.0] }
]
```

As long as the column name is `vector` then we are ok.  The default value of `vectorColumns` is `["vector"]` and so any column named `vector` will be made into a fixed size list.

Also, this change should not be changing any existing behavior so as long as those examples were working before they should still be working after this change.